ports/stm32/machine_timer.c

```diff
@@ -56,7 +56,7 @@ STATIC mp_obj_t machine_timer_init_helper(machine_timer_obj_t *self, size_t n_ar
     if (args[ARG_freq].u_obj != mp_const_none) {
         // Frequency specified in Hz
         #if MICROPY_PY_BUILTINS_FLOAT
-        self->delta_ms = 1000 / mp_obj_get_float(args[ARG_freq].u_obj);
+        self->delta_ms = 1000 / (uint32_t)mp_obj_get_float(args[ARG_freq].u_obj);
```

Comment:
This changes behaviour to do integer instead of float division.  Should be `(uint32_t)(MICROPY_FLOAT_CONST(1000.0) / mp_obj_get_float(...))`.  Hopefully that gives equivalent binaries.